gpt_bpe_test.go

```diff
@@ -907,80 +1006,112 @@ func TestPythiaRemoteDownloadTokenizer(t *testing.T) {
 	}
 }
 
+func TestLlama3RemoteDownloadTokenizer(t *testing.T) {
+	// Tests the ability to download a tokenizer from a remote model
+	// and use it to encode and decode strings
+	modelId := "Groq/Llama-3-Groq-8B-Tool-Use" // Original Llama3 model is gated
+	destPath := "./TestLlama3RemoteDownloadTokenizer"
+	defer os.RemoveAll(destPath)
+	encoderLlama3, err := NewEncoder(modelId)
+	if err != nil {
+		t.Errorf("Error creating encoder: %v", err)
+	}
+
+	// Attempt to tokenize
+	testString := "The fox jumped over the hare.\nThe turtle is faster than the hare."
+
+	// Encode the string
+	encoded := encoderLlama3.Encode(&testString)
+	// Check that the encoded string is the same as the expected - 128009 is 128001 in the original Llama3 model
+	expected := Tokens{128000, 791, 39935, 27096, 927, 279, 96018, 627, 791, 37189, 374, 10819, 1109, 279, 96018, 13, 128009}
+	if !assert.Equal(t, expected, *encoded) {
+		t.Errorf("Expected: %v\nActual: %v", expected, *encoded)
+	}
+}
+
+func TestMistralRemoteDownloadTokenizer(t *testing.T) {
+	// Tests the ability to download a tokenizer from a remote model
+	// and use it to encode and decode strings
+	modelId := "Open-Orca/Mistral-7B-OpenOrca"
+	//destPath := "./TestMistralRemoteDownloadTokenizer"
+	//defer os.RemoveAll(destPath)
+	encoderMistral, err := NewEncoder(modelId)
+	if err != nil {
+		t.Errorf("Error creating encoder: %v", err)
+	}
+
+	// Attempt to tokenize
+	testString := "The fox jumped over the hare.\nThe turtle is faster than the hare."
+
+	// Encode the string
+	encoded := encoderMistral.Encode(&testString)
+	// Check that the encoded string is the same as the expected - Reference from python's transformers lib
+	expected := Tokens{1, 1014, 285, 1142, 14949, 754, 272, 295, 492, 28723, 13, 1014, 261, 3525, 291, 349, 9556, 821, 272, 295, 492, 28723}
+	if !assert.Equal(t, expected, *encoded) {
+		t.Errorf("Expected: %v\nActual: %v", expected, *encoded)
+	}
+}
+
 func TestGPTDecoder_Decode(t *testing.T) {
 	// TBD
 }
 
 func TestRankPairs(t *testing.T) {
 }
 
-func TestModelDownload(t *testing.T) {
+func downloadModel(modelId string, destPath string) error {
 	// Download the model
-	modelId := "gpt2"
-	destPath := "./TestModelDownload"
 	destPathPTR := &destPath
-
 	rsrcType, hfApiToken := resources.RESOURCETYPE_TRANSFORMERS, os.Getenv("HF_API_TOKEN")
```

Comment:
This might be out of scope so I'm just bringing attention to it, HF updated the token env var name to `HF_TOKEN`.